packages/bull-vault/src/BullStrategy.sol

```diff
@@ -145,17 +142,20 @@ contract BullStrategy is ERC20, LeverageBull {
         }
 
         (uint256 ethInCrab, uint256 squeethInCrab) = _getCrabVaultDetails();
+        // deposit eth into leverage component and borrow USDC
         (, uint256 usdcBorrowed, uint256 _totalWethInEuler) = _leverageDeposit(
             msg.value, bullToMint, share, ethInCrab, squeethInCrab, IERC20(crab).totalSupply()
         );
 
         require(_totalWethInEuler <= strategyCap, "BS2");
 
         IERC20(usdc).transfer(msg.sender, usdcBorrowed);
+
+        emit Deposit(msg.sender, _crabAmount);
     }
 
     /**
-     * @notice withdraw ETH from crab and euler by providing wPowerPerp, bull token and USDC to repay debt
+     * @notice withdraw ETH from crab and Euler by providing wPowerPerp, bull token, and USDC to repay debt
```

Comment:
maybe should note the fact that withdrawals are in ETH but deposits are in crab, in fairness maybe not the best decision :)

Reply:
added descriptions